drivers/sensor/ite_tach_it8xxx2/tach_ite_it8xxx2.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2021 ITE Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_tach
+
+/**
+ * @file
+ * @brief ITE it8xxx2 tachometer sensor module driver
+ *
+ * This file contains a driver for the tachometer sensor module which contains
+ * two independent counters (F1TL/MRR and F2TL/MRR). The content of the
+ * Tachometer Reading Register is still update based on the sampling counter
+ * that samples the tachometer input (T0A, T0B, T1A or T1B pins).
+ * The following is block diagram of this module:
+ *
+ *                                    Sample Rate = TACH_FREQ / 128
+ *                                                   |
+ *                            |        Tachometer 0  |                T0A (GPD6)
+ *                            |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>| F1TL/MRR  |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T0B (GPJ2)
+ *                            |       in sample rate
+ *                            |       period
+ *           +-----------+    |
+ * Crystal-->| Prescaler |--->|        Tachometer 1                   T1A (GPD7)
+ * 32.768k   +-----------+    |             |          +-----------+   |
+ *                            |       +-----+-----+    |   _   _   |<--+
+ *                            |------>| F2TL/MRR  |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T1B (GPJ3)
+ *                            |       in one second
+ *                            |       period
+ *                            |
+ *
+ * Based on the counter value, we can compute the current RPM of external signal
+ * from encoders.
+ */
+
+#include <device.h>
+#include <drivers/pinmux.h>
+#include <drivers/sensor.h>
+#include <dt-bindings/sensor/it8xxx2_tach.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tach_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * NOTE: The PWM output maximum is 324Hz in EC LPM, so if we need fan to work
+ *       then don't let EC enter LPM.
+ */
+#define TACH_FREQ		EC_FREQ
+
+/* Device config */
+struct tach_alt_cfg {
+	/* Pinmux control device structure */
+	const struct device *pinctrls;
```

Comment:
not related to this PR, but be aware that drivers should start moving to `pinctrl`, see https://github.com/zephyrproject-rtos/zephyr/issues/39740

Reply:
Ack